raiden/transfer/channel.py

```diff
@@ -1693,22 +1802,44 @@ def handle_action_set_fee(
 
 def handle_action_withdraw(
     channel_state: NettingChannelState,
-    withdraw: ActionChannelWithdraw,
+    action_withdraw: ActionChannelWithdraw,
     pseudo_random_generator: random.Random,
+    block_number: BlockNumber,
 ) -> TransitionResult[NettingChannelState]:
-    is_valid, msg = is_valid_action_withdraw(channel_state, withdraw)
+    events: List[Event] = list()
+    is_valid, msg = is_valid_action_withdraw(channel_state, action_withdraw)
 
     if is_valid:
-        channel_state.our_state.total_withdraw = withdraw.total_withdraw
-        events: List[Event] = send_withdraw_request(
+        nonce = get_next_nonce(channel_state.our_state)
+        # Calculate the withdraw's expiry exactly the same as we
+        # do for lock.
+        # That value defines how many blocks it takes for a transaction to be
+        # mined under congestion.
+        # An expiration lower than 1 * reveal_timeout  means that we are requesting
+        # a withdraw that will fail under congestion.
+        # A value larger then 2 * reveal_timeout means Raiden would be slower
+        # than the blockchain.
+        expiration = get_initial_lock_expiration(
+            block_number=block_number, reveal_timeout=channel_state.reveal_timeout
+        )
+        withdraw_state = WithdrawState(
+            total_withdraw=action_withdraw.total_withdraw, nonce=nonce, expiration=expiration
+        )
+        channel_state.our_state.total_withdraw = action_withdraw.total_withdraw
+        channel_state.our_state.nonce = nonce
```

Comment:
please, don't split the state update into multiple functions. This should be in `send_withdraw_request`, that function should take a valid channel state and produce a valid state change. Right now the channel stae given to `send_withdraw_request` is invalid (the nonce is updated but the event for the message was not produced)

Reply:
I replied to an earlier comment you made about this same issue [here](https://github.com/raiden-network/raiden/pull/4130#discussion_r295334344). 

> This was done because in some of our tests, we call create_sendlockedtransfer to create a transfer by the partner (to simulate receiving a transfer from our partner)
> in this case, the nonce of the partner will be increased in the create_send* functions which will fail the tests.

So what was done here is consistent:
1. All callers of `send_*` will update the state
2. `send_*` functions will/should only produce a message.

I'll see if i can change this and fix the tests